zipkin/propagation.go

```diff
@@ -24,12 +24,21 @@ import (
 )
 
 // Propagator is an Injector and Extractor
-type Propagator struct{}
+type Propagator struct {
+	baggagePrefix string
+}
 
 // NewZipkinB3HTTPHeaderPropagator creates a Propagator for extracting and injecting
-// Zipkin HTTP B3 headers into SpanContexts.
+// Zipkin HTTP B3 headers into SpanContexts. Baggage is by default enabled and uses prefix
+// 'baggage-'.
 func NewZipkinB3HTTPHeaderPropagator() Propagator {
```

Comment:
:+1:  it's nice design

Reply:
There are various styles how this can be done. I have changed it slightly in the last commit.

Reply:
Go experts advise against using functional options and use option structs, because structs compose better (as evidenced by our duplication of tracer and config options).

The reason we traditionally used options is in cases where the existing API did not have configurability, in which case adding a vararg option list is backwards compatible.

Reply:
Is there a resource on this? I remember someone talking about this but never read anything about it

Reply:
@yurishkuro do you it's better to pass option struct to the "constructor" function (or whatever it's called).

Reply:
I think for new code it might make sense to pass a struct with options.